src/components/offcanvas/OffcanvasHeader.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {omit} from 'ramda';
+import {default as RBOffcanvasHeader} from 'react-bootstrap/OffcanvasHeader';
+import {default as RBOffcanvasTitle} from 'react-bootstrap/OffcanvasTitle';
+
+/**
+ * Add a header to any offcanvas.
+ */
+const OffcanvasHeader = props => {
+  const {title, class_name, close_button, ...otherProps} = props;
+  return (
+    <RBOffcanvasHeader
+      className={class_name}
+      closeButton={close_button}
```

Comment:
It would be good if the close button could be used to trigger callbacks. Perhaps the header should be incorporated into the Offcanvas component itself so that close events can be registered there?

Reply:
There are multiple ways of dismissing the offcanvas (`is_open` property, escape key, clicking backdrop and the close button itself). In the main `Offcanvas` component, we capture these together so when it is dismissed, the `is_open` property changes. Would it be sufficient to use this to trigger callbacks, or should there be additional variations to capture each possible way of dismissing it?

Worth also mentioning that the Offcanvas component in react-bootstrap has a number of events itself, so we may potentially want to take advantage of some of these. I included them in a comment in the code, but putting it here to keep things together (and allow me to remove from the code itself):

- enforceFocus
- onEnter
- onEntered
- onEntering
- onEscapeKeyDown
- onExit
- onExited
- onExiting
- onShow
- restoreFocus
- restoreFocusOptions

In addition, it has a `container` property, which I haven't implemented.